common/persistence/elasticsearch/visibility_store.go

```diff
@@ -1197,3 +1197,24 @@ func parseHHMMSSDuration(d string) (int64, error) {
 
 	return hours*int64(time.Hour) + minutes*int64(time.Minute) + seconds*int64(time.Second) + nanos, nil
 }
+
+func detailedErrorMessage(err error) string {
```

Comment:
could you add a unit test for this method.

Reply:
Yes, I was about to do it. Adding.